File: framework/languages/dotnet/e2e/SpotService/Client/Scenarios/SmA6Scenario.cs
using SpotService.Client.Support;
using SpotService.Shared;
using Zlink.HttpClient;

namespace SpotService.Client.Scenarios;

internal static class SmA6Scenario
{
    public static async Task RunAsync(ZLinkHttpClient playA)
    {
        var spotRid = $"spot-sm-a6-{Guid.NewGuid():N}";
        var created = (await playA.Post("/spot/create")
            .Body(new CreateSpotReq(spotRid))
            .SubmitAsync<CreateSpotReply>()).Body;
        ScenarioAssert.That(created.SpotRid == spotRid && created.NodeRid == "play-a",
            "SM-A6 lifecycle spot was not created on play-a.");
        var closeReply = (await playA.Post("/spot/close")
            .Body(new CloseSpotReq(spotRid))
            .SubmitAsync<CloseSpotReply>()).Body;
        ScenarioAssert.That(closeReply.Closed, "SM-A6 did not close the lifecycle spot.");
        var expectedEvidence = new[]
        {
            $"spot-initialize|rid=play-a|spot={spotRid}",
            $"spot-closing|rid=play-a|spot={spotRid}"
        };
        var evidence = (await playA.Post("/evidence/wait")
            .Body(new EvidenceWaitRequest(expectedEvidence))
            .SubmitAsync<string[]>()).Body;
        ScenarioAssert.That(
            expectedEvidence.All(expected => evidence.Any(line => line.Contains(expected, StringComparison.Ordinal))),
            "SM-A6 evidence mismatch.");
        Console.WriteLine("operation SpotService.sm-a6 passed");
    }
}

File: framework/languages/node/e2e/SpotService/Client/Scenarios/sm-a6-scenario.ts
// SM-A6: spot lifecycle (initialize·close) 시나리오를 검증한다.
import type { CloseSpotRes, CloseSpotReq, CreateSpotRes, CreateSpotReq, EvidenceWaitReq } from '../../Shared/messages';
import type { ClientOptions } from '../Support/client-options';
import { postJson } from '../../../http-client';
import { ensure } from '../Support/scenario-assert';

export async function runSmA6(options: ClientOptions): Promise<void> {
  const spotRid = `spot-sm-a6-${Date.now().toString(36)}`;
  const created = await postJson<CreateSpotRes>(options.playAUrl, '/spot/create', {
    spotRid
  } satisfies CreateSpotReq);
  ensure(created.spotRid === spotRid, 'SM-A6 lifecycle spot was not created.');
  ensure(created.nodeRid === 'play-a', 'SM-A6 lifecycle spot was created on the wrong node.');

  const closed = await postJson<CloseSpotRes>(options.playAUrl, '/spot/close', {
    spotRid
  } satisfies CloseSpotReq);
  ensure(closed.closed, 'SM-A6 did not close the lifecycle spot.');

  const expected = [
    `spot-initialize|rid=play-a|spot=${spotRid}`,
    `spot-closing|rid=play-a|spot=${spotRid}`
  ];
  const evidence = await postJson<string[]>(options.playAUrl, '/evidence/wait', {
    containsAll: expected,
    timeoutMilliseconds: 10000
  } satisfies EvidenceWaitReq);
  ensure(
    expected.every((marker) => evidence.some((line) => line.includes(marker))),
    'SM-A6 lifecycle evidence mismatch.'
  );

  console.log('scenario SM-A6 passed');
}
